test/fuzz/transformation_add_parameter_test.cpp

```diff
@@ -240,10 +239,343 @@ TEST(TransformationAddParameterTest, BasicTest) {
+         %35 = OpVariable %9 Function
+         %36 = OpVariable %9 Function
+               OpStore %24 %25
+               OpStore %27 %28
+         %29 = OpFunctionCall %2 %6
+               OpStore %30 %18
+         %33 = OpLoad %8 %30
+               OpStore %32 %33
+         %34 = OpFunctionCall %8 %12 %32
+               OpStore %31 %34
+         %37 = OpLoad %8 %31
+               OpStore %36 %37
+         %38 = OpFunctionCall %8 %12 %36
+               OpStore %35 %38
+         ; %39 = OpFunctionCall %2 %14
+               OpReturn
+               OpFunctionEnd
+          %6 = OpFunction %2 None %3
+          %7 = OpLabel
+               OpReturn
+               OpFunctionEnd
+         %12 = OpFunction %8 None %10
+         %11 = OpFunctionParameter %9
+         %13 = OpLabel
+         %17 = OpLoad %8 %11
+         %19 = OpIAdd %8 %17 %18
+               OpReturnValue %19
+               OpFunctionEnd
+         %14 = OpFunction %2 None %3
+         %15 = OpLabel
+               OpReturn
+               OpFunctionEnd
+    )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  FactManager fact_manager;
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(&fact_manager,
+                                               validator_options);
+
+  // Bad: Id 19 is not available in the caller that has id 25.
+  TransformationAddParameter transformation_bad_1(12, 50, 8,
+                                                  {{{34, 19}, {38, 19}}}, 51);
+
+  ASSERT_FALSE(
+      transformation_bad_1.IsApplicable(context.get(), transformation_context));
+
+  // Bad: Id 8 does not have a type.
+  TransformationAddParameter transformation_bad_2(12, 50, 8,
+                                                  {{{34, 8}, {38, 8}}}, 51);
+
+  ASSERT_FALSE(
+      transformation_bad_2.IsApplicable(context.get(), transformation_context));
+
+  // Bad: Types of id 25 and id 18 are different.
+  TransformationAddParameter transformation_bad_3(12, 50, 22,
+                                                  {{{34, 25}, {38, 18}}}, 51);
+  ASSERT_FALSE(
+      transformation_bad_3.IsApplicable(context.get(), transformation_context));
+
+  // Function with id 14 does not have any callers.
+  // Bad: Id 18 is not a vaild type.
+  TransformationAddParameter transformation_bad_4(14, 50, 18, {{}}, 51);
+  ASSERT_FALSE(
+      transformation_bad_4.IsApplicable(context.get(), transformation_context));
+
+  // Function with id 14 does not have any callers.
+  // Bad:  Id 3 refers to OpTypeVoid, which is not supported.
+  TransformationAddParameter transformation_bad_6(14, 50, 3, {{}}, 51);
+  ASSERT_FALSE(
+      transformation_bad_6.IsApplicable(context.get(), transformation_context));
+}
+
+TEST(TransformationAddParameterTest, PointerTypeTest) {
```

Comment:
The tests look good, but please add global pointer parameters with storage class Private and Workgroup. Feel free to do this in a new test (or 2 new tests), and no need to cover the many different failure cases. But we at least need the success cases, as this would have caught the mistake of trying to give a workgroup variable an initializer. 

You might want a compute shader example from here:

https://github.com/google/graphicsfuzz/tree/master/shaders/src/main/glsl/samples/compute/320es